openbas-framework/src/main/java/io/openbas/asset/AssetGroupService.java

```diff
@@ -94,7 +94,16 @@ private List<AssetGroup> computeDynamicAssets(@NotNull final List<AssetGroup> as
         assetGroups.forEach(assetGroup -> {
             if (!isEmptyFilterGroup(assetGroup.getDynamicFilter())) {
                 Predicate<Object> filters = computeFilterGroupRuntime(assetGroup.getDynamicFilter());
-                assetGroup.setDynamicAssets(assets.stream().filter(filters).toList());
+                // Filters for dynamic assets are applicable only to endpoints
+                List<Asset> endpointAssets = assets.stream()
+                    .filter(asset -> "Endpoint".equals(asset.getType()))
+                    .toList();
```

Comment:
```suggestion
                    .filter(filters)
                    .toList();
```
Why not chain the filters instead of using a temporary variable ?

Reply:
Sure! I'll change it!